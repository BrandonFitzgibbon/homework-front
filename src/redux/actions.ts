import { IEntry } from '../types/interfaces'

export enum Type {
    POST_ALL_ENTRY = "POST_ALL_ENTRY",
    POST_CURRENT_ENTRY = "POST_CURRENT_ENTRY",
    SET_ALL_ENTRIES = "SET_ALL_ENTRIES",
    SET_CURRENT_ENTRIES = "SET_CURRENT_ENTRIES",
    CHANGE_NAME = "CHANGE_NAME",
    CHANGE_TARGET = "CHANGE_TARGET"
}

export function postAllEntry(entry: IEntry) {
    return { type: Type.POST_ALL_ENTRY, entries: [entry] }
}

export function postCurrentEntry(entry: IEntry) {
    return { type: Type.POST_CURRENT_ENTRY, entries: [entry] }
}

export function setAllEntries(entries: IEntry[]) {
    return { type: Type.SET_ALL_ENTRIES, entries}
}

export function setCurrentEntries(entries: IEntry[]) {
    return { type: Type.SET_CURRENT_ENTRIES, entries}
}

export function changeName(name: string) {
    return { type: Type.CHANGE_NAME, name }
}

export function changeTarget(target_id: string) {
    return { type: Type.CHANGE_TARGET, target_id}
}