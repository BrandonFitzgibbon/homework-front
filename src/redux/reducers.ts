import { combineReducers } from 'redux'
import { IEntry } from '../types/interfaces'
import { Type } from './actions'

function allEntries(state: IEntry[] = [], action: { type: Type, entries: IEntry[] }) : IEntry[] {
    switch(action.type) {
        case Type.POST_ALL_ENTRY:
            return [
                ...state,
                ...action.entries
            ]
        case Type.SET_ALL_ENTRIES:
            return action.entries
        default:
            return state
    }
}

function currentEntries(state: IEntry[] = [], action: { type: Type, entries: IEntry[] }) : IEntry[] {
    switch(action.type) {
        case Type.POST_CURRENT_ENTRY:
            return [
                ...state,
                ...action.entries
            ]
        case Type.SET_CURRENT_ENTRIES:
            return action.entries
        default:
            return state
    }
}

function currentName(state: string = '', action: { type: Type, name: string}) : string {
    switch(action.type) {
        case Type.CHANGE_NAME:
            return action.name
        default:
            return state
    }
    
}

function currentTarget(state: string = '', action: { type: Type, target_id: string}) : string {
    switch(action.type) {
        case Type.CHANGE_TARGET:
            return action.target_id
        default:
            return state
    }
}

const app = combineReducers({
    allEntries,
    currentEntries,
    currentName,
    currentTarget
})

export default app

