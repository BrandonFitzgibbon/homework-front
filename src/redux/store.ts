import { createStore } from 'redux'
import app from './reducers'

export const store = createStore(app, (window as any).__REDUX_DEVTOOLS_EXTENSION__ && (window as any).__REDUX_DEVTOOLS_EXTENSION__())