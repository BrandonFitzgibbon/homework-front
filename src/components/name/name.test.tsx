import React from 'react'
import Name from './index'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'

jest.mock('../../api/entries')
jest.mock('../../api/replies')

test('name component states', async() => {
    let name : ReactTestRenderer | undefined

    act(() => {
        name = renderer.create(
            <Provider store={store}>
                <Name />
            </Provider>
        )
    })

    const snap = () => {
        if (name) {
            let tree = name.toJSON()
            expect(tree).toMatchSnapshot()
        }
    }

    let textInput
    if (name) {
        textInput = name.root.findAllByType('input').find(i => i.props.placeholder === "Name")
    }

    //initial state
    snap()

    //add some text
    if (textInput) {
        textInput.props.onChange({target:{value:"Test"}})
    }

    snap()
})