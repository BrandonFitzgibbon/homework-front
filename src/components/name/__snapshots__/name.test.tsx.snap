// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`name component states 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    disabled={true}
    type="submit"
    value="Submit"
  />
  <p>
    Please enter your name to post or reply
  </p>
</form>
`;

exports[`name component states 2`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    value="Test"
  />
  <input
    disabled={false}
    type="submit"
    value="Submit"
  />
  <p>
    Please enter your name to post or reply
  </p>
</form>
`;
